### ISSUE TITLE
[REVIEW]: Aero Python: classical aerodynamics of potential flow using Python

### ISSUE THREAD

**Issue by whedon** [openjournals/jose-reviews#45]
**Submitting author:** @labarba (<a href="http://orcid.org/0000-0001-5812-2711">Lorena A. Barba</a>)
**Repository:** <a href="https://github.com/barbagroup/AeroPython" target ="_blank">https://github.com/barbagroup/AeroPython</a>
**Version:** v1.0
**Editor:** @kyleniemeyer
**Reviewer:** @weymouth, @Juanlu001
**Archive:** <a href="https://doi.org/10.6084/m9.figshare.1004727.v4" target="_blank">10.6084/m9.figshare.1004727.v4</a>

## Status

[![status](http://jose.theoj.org/papers/b679b34c976beec0bc64807bf087a468/status.svg)](http://jose.theoj.org/papers/b679b34c976beec0bc64807bf087a468)

Status badge code:

```
HTML: <a href="http://jose.theoj.org/papers/b679b34c976beec0bc64807bf087a468"><img src="http://jose.theoj.org/papers/b679b34c976beec0bc64807bf087a468/status.svg"></a>
Markdown: [![status](http://jose.theoj.org/papers/b679b34c976beec0bc64807bf087a468/status.svg)](http://jose.theoj.org/papers/b679b34c976beec0bc64807bf087a468)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/barbagroup/AeroPython" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@weymouth & @Juanlu001, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/jose-reviews/invitations

The reviewer guidelines are available here: https://jose.theoj.org/about#reviewer_guidelines. Any questions/concerns please let @kyleniemeyer know.

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSE conflict of interest policy](https://github.com/openjournals/jose/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSE code of conduct](https://jose.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source for this learning module available at the <a target="_blank" href="https://github.com/barbagroup/AeroPython">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of a standard license? (OSI-approved for code, Creative Commons for content)
- [x] **Version:** v1.0
- [x] **Authorship:** Has the submitting author (@labarba) made visible contributions to the module? Does the full list of authors seem appropriate and complete?

### Documentation

- [x] **A statement of need:** Do the authors clearly state the need for this module and who the target audience is?
- [x] **Installation instructions:** Is there a clearly stated list of dependencies?
- [x] **Usage:** Does the documentation explain how someone would adopt the module, and include examples of how to use it?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the module 2) Report issues or problems with the module 3) Seek support

### Pedagogy / Instructional design (Work-in-progress: reviewers, please comment!)

- [x] **Learning objectives:** Does the module make the learning objectives plainly clear? (We don't require explicitly written learning objectives; only that they be evident from content and design.)
- [x] **Content scope and length:** Is the content substantial for learning a given topic? Is the length of the module appropriate?
- [x] **Pedagogy:** Does the module seem easy to follow? Does it observe guidance on cognitive load? (working memory limits of 7 +/- 2 chunks of information)
- [x] **Content quality:** Is the writing of good quality, concise, engaging? Are the code components well crafted? Does the module seem complete?
- [x] **Instructional design:** Is the instructional design deliberate and apparent? For example, exploit worked-example effects; effective multi-media use; low extraneous cognitive load.

### JOSE paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Does the paper clearly state the need for this module and who the target audience is?
- [x] **Description:** Does the paper describe the learning materials and sequence?
- [x] Does it describe how it has been used in the classroom or other settings, and how someone might adopt it?
- [x] Could someone else teach with this module, given the right expertise?
- [ ] Does the paper tell the "story" of how the authors came to develop it, or what their expertise is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSE conflict of interest policy](https://github.com/openjournals/jose/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSE code of conduct](https://jose.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source for this learning module available at the <a target="_blank" href="https://github.com/barbagroup/AeroPython">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of a standard license? (OSI-approved for code, Creative Commons for content)
- [x] **Version:** v1.0
- [x] **Authorship:** Has the submitting author (@labarba) made visible contributions to the module? Does the full list of authors seem appropriate and complete?

### Documentation

- [ ] **A statement of need:** Do the authors clearly state the need for this module and who the target audience is?
- [x] **Installation instructions:** Is there a clearly stated list of dependencies?
- [x] **Usage:** Does the documentation explain how someone would adopt the module, and include examples of how to use it?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the module 2) Report issues or problems with the module 3) Seek support

### Pedagogy / Instructional design (Work-in-progress: reviewers, please comment!)

- [x] **Learning objectives:** Does the module make the learning objectives plainly clear? (We don't require explicitly written learning objectives; only that they be evident from content and design.)
- [x] **Content scope and length:** Is the content substantial for learning a given topic? Is the length of the module appropriate?
- [x] **Pedagogy:** Does the module seem easy to follow? Does it observe guidance on cognitive load? (working memory limits of 7 +/- 2 chunks of information)
- [x] **Content quality:** Is the writing of good quality, concise, engaging? Are the code components well crafted? Does the module seem complete?
- [x] **Instructional design:** Is the instructional design deliberate and apparent? For example, exploit worked-example effects; effective multi-media use; low extraneous cognitive load.

### JOSE paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Does the paper clearly state the need for this module and who the target audience is?
- [x] **Description:** Does the paper describe the learning materials and sequence?
- [x] Does it describe how it has been used in the classroom or other settings, and how someone might adopt it?
- [x] Could someone else teach with this module, given the right expertise?
- [x] Does the paper tell the "story" of how the authors came to develop it, or what their expertise is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @weymouth, it looks like you're currently assigned as the reviewer for this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/jose-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/jose-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/jose-papers/blob/jose.00045/jose.00045/10.21105.jose.00045.pdf)

**Comment by kyleniemeyer** [comment]
👋 @weymouth @Juanlu001 @labarba the actual review will take place here; please note the separate review checklists above

**Comment by Juanlu001** [comment]
Hi all - sorry this is taking me so long. I will try to start the review next week, or the following one at most.

**Comment by labarba** [comment]
Looking forward to your review comments, @weymouth and @Juanlu001 !

**Comment by weymouth** [comment]
I think this is a great repository and a model for other similar projects. The only things I notice are 

1. The write-up doesn't make the development "story" clear. 
1. The usage of the code in the repository is well explained. Would it be helpful to explain how this ties in with the other parts of the class? For example, is this supposed to be their first time seeing any of these concepts? What do they do with this experience afterward?
1. I don't see the version number. (Which I only mention because of the checklist.)

**Comment by mesnardo** [comment]
Hi @weymouth,

Thank you for your review of AeroPython! We'll work on your suggestions.

(@Juanlu001, looking forward to your review.)

**Comment by kyleniemeyer** [comment]
Hi @Juanlu001, just wanted to check in on the status of your review.

@mesnardo have you made changes according to @weymouth's suggestions?

**Comment by mesnardo** [comment]
@kyleniemeyer We are waiting for the review of @Juanlu001 before making the suggested modifications.

**Comment by labarba** [comment]
Hi @Juanlu001 — we're eager to get  your review of the AeroPython module, to make a revision and get it to publication-ready status. Would you be able to get to this soon?

**Comment by Juanlu001** [comment]
Sorry for the super long delay. On it!

**Comment by Juanlu001** [comment]
It was quick because I already knew the course from the time it was released :)

Non-blocking comments:

* There is no version number, so I left that unchecked
* There is a statement of need in the paper but not in the repository itself - I understand that there are two checkboxes for this reason, so I left the first unchecked as well
* Perhaps this is a bit presumptuous on my side, but I would add a little footnote somewhere stating somethink like "not to be confused with another Python course for aeronautical engineers in Spanish language, with the same name but different content" (refs: https://github.com/AeroPython/Curso_AeroPython/, https://twitter.com/LorenaABarba/status/464041427169583104)

**Comment by mesnardo** [comment]
Thank you @weymouth and @Juanlu001 for your feedback!

Modifications based on your suggestions are available in the branch [`jose_revision`](https://github.com/barbagroup/AeroPython/tree/jose_revision) ; their is an open PR ([#44](https://github.com/barbagroup/AeroPython/pull/44)) to merge these commits in the `master` branch.

* We added a version number to the GitHub repository; the GitHub release will be done once the revision is finished.
* We added a note at the bottom of the README to mention the other [AeroPython](https://github.com/AeroPython/Curso_AeroPython/) repository.
* We modified the write-up to mention how AeroPython can be used as a full-semester course.

**Comment by labarba** [comment]
Regarding the Statement of Need, the JOSE editors decided it would only be required in the paper (not the documentation), but unfortunately the checklists haven't been updated yet!

**Comment by labarba** [comment]
👋  @weymouth, @Juanlu001 — I think we're ready with this revision. Would you take a final look? The editor will need your recommendation to accept.

**Comment by weymouth** [comment]
Looks good to me!

**Comment by Juanlu001** [comment]
And to me! :+1:

**Comment by labarba** [comment]
Thank you, @weymouth and @Juanlu001 — we appreciate your time reviewing this submission!

@kyleniemeyer — we await your final checks for publication.

**Comment by kyleniemeyer** [comment]
@labarba will do those shortly

**Comment by kyleniemeyer** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/jose-papers/blob/jose.00045/jose.00045/10.21105.jose.00045.pdf)

**Comment by kyleniemeyer** [comment]
@labarba @mesnardo  a few minor comments on the paper itself:

- Could you add full affiliation details (city, state, country)?
- I believe the final bullet in the list of lessons should start with "Assignment 3"
- I just submitted a small fix to the paper near the end; multiple references in the same square brackets should be separated by semicolons: barbagroup/AeroPython#45
- the Catrambone reference is missing the DOI: 10.1037/0096-3445.127.4.355
- the Margulieux reference is missing the DOI: 10.1080/08993408.2016.1144429

Once you make those fixes, please archive the repo and provide the DOI. Almost done!

**Comment by mesnardo** [comment]
> @labarba @mesnardo a few minor comments on the paper itself:
> 
> * Could you add full affiliation details (city, state, country)?

Done in [35cf51d](https://github.com/barbagroup/AeroPython/commit/35cf51df75dae985cfd2dae20bb1f68b71a0a51b).

> * I believe the final bullet in the list of lessons should start with "Assignment 3"

Done in [8209e1a](https://github.com/barbagroup/AeroPython/commit/8209e1a336f02e9ee7ce2821673515b803485b09).

> * I just submitted a small fix to the paper near the end; multiple references in the same square brackets should be separated by semicolons: [barbagroup/AeroPython#45](https://github.com/barbagroup/AeroPython/pull/45)

Thank you.

> * the Catrambone reference is missing the DOI: 10.1037/0096-3445.127.4.355
> * the Margulieux reference is missing the DOI: 10.1080/08993408.2016.1144429

Added mission DOIs in [64b97fc](https://github.com/barbagroup/AeroPython/commit/64b97fc62b202058854033d74e956b18e190e70e).

> 
> Once you make those fixes, please archive the repo and provide the DOI. Almost done!

**Comment by kyleniemeyer** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/jose-papers/blob/jose.00045/jose.00045/10.21105.jose.00045.pdf)

**Comment by kyleniemeyer** [comment]
@mesnardo looks like those DOIs were not actually added properly (used a bibdesk URL field rather than DOI field); I fixed in PR barbagroup/AeroPython#46

That should be the final edit necessary; please merge and then deposit for a DOI.

**Comment by labarba** [comment]
…and sneaked in an en-dash, I see 😁 

PR merged. Thanks!

**Comment by kyleniemeyer** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/jose-papers/blob/jose.00045/jose.00045/10.21105.jose.00045.pdf)

**Comment by kyleniemeyer** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by kyleniemeyer** [comment]
@labarba @mesnardo I'm not sure why the reference checking isn't working, but this looks good to go to me. Could you archive and provide me with the DOI?

**Comment by labarba** [comment]
@whedon set v1.0 as version

**Comment by whedon** [comment]
OK. v1.0 is the version.

**Comment by labarba** [comment]
👋 @weymouth, @Juanlu001 — We have now released v1.0 and updated the version number here, so you can tick off that item in your checklist. Many thanks!

**Comment by labarba** [comment]
@whedon set 10.6084/m9.figshare.1004727.v4 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.6084/m9.figshare.1004727.v4" target="_blank">10.6084/m9.figshare.1004727.v4</a> is the archive.

**Comment by kyleniemeyer** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #45 with the following error: 

 /app/vendor/ruby-2.4.4/lib/ruby/2.4.0/find.rb:43:in `block in find': No such file or directory - tmp/45 (Errno::ENOENT)
	from /app/vendor/ruby-2.4.4/lib/ruby/2.4.0/find.rb:43:in `collect!'
	from /app/vendor/ruby-2.4.4/lib/ruby/2.4.0/find.rb:43:in `find'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-a1723d160bb6/lib/whedon/processor.rb:57:in `find_paper_paths'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-a1723d160bb6/bin/whedon:73:in `compile'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-a1723d160bb6/bin/whedon:116:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `load'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `<main>'

**Comment by kyleniemeyer** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/jose-papers/blob/jose.00045/jose.00045/10.21105.jose.00045.pdf)

**Comment by kyleniemeyer** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/jose-papers/pull/33

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/jose-papers/pull/33, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by kyleniemeyer** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
I'm sorry @kyleniemeyer, I'm afraid I can't do that. That's something only editor-in-chiefs are allowed to do.

**Comment by labarba** [comment]
Yikes!

**Comment by kyleniemeyer** [comment]
@labarba hmm, I think I am supposed to have that power...

**Comment by labarba** [comment]
On JOSS, yes. But here, we never set up an EiC team.

**Comment by kyleniemeyer** [comment]
@labarba Ah. Well, we can figure that out offline. In the meantime, JOSE EiC, I'd like to recommend this submission be accepted 😄

**Comment by labarba** [comment]
I was just fiddling to create a new team (which I did), but can't find how to give it Admin powers. Gah.

**Comment by labarba** [comment]
I'll just go ahead and publish this.

**Comment by labarba** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSE!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/jose-papers/pull/34
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/jose.00045](https://doi.org/10.21105/jose.00045)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? notify your editorial technical team...

**Comment by kyleniemeyer** [comment]
Congrats to @labarba and @mesnardo on your submissions's publication in JOSE! Thanks to @weymouth and @Juanlu001 for reviewing!

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://jose.theoj.org/papers/10.21105/jose.00045/status.svg)](https://doi.org/10.21105/jose.00045)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/jose.00045">
  <img src="https://jose.theoj.org/papers/10.21105/jose.00045/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://jose.theoj.org/papers/10.21105/jose.00045/status.svg
   :target: https://doi.org/10.21105/jose.00045
```
This is how it will look in your documentation:

[![DOI](https://jose.theoj.org/papers/10.21105/jose.00045/status.svg)](https://doi.org/10.21105/jose.00045)

**We need your help!**

Journal of Open Source Education is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. 
- Making a small donation to support our running costs here: https://numfocus.salsalabs.org/donate-to-jose

**Comment by labarba** [comment]
https://twitter.com/JOSE_TheOJ/status/1129425838406082560

**Comment by mesnardo** [comment]
Yay! Many thanks to @kyleniemeyer, @Juanlu001, and @weymouth for all the suggestions and modifications provided!

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1080/08993408.2016.1144429 is OK
- 10.1037/0096-3445.127.4.355 is OK
- 10.1037/edu0000018 is OK
- 10.1016/j.learninstruc.2006.02.005 is OK
- 10.1007/978-3-642-84010-4_1 is OK
- 10.21105/jose.00021 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1080/08993408.2016.1144429 is OK
- 10.1037/0096-3445.127.4.355 is OK
- 10.1037/edu0000018 is OK
- 10.1016/j.learninstruc.2006.02.005 is OK
- 10.1007/978-3-642-84010-4_1 is OK
- 10.21105/jose.00021 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1080/08993408.2016.1144429 is OK
- 10.1037/0096-3445.127.4.355 is OK
- 10.1037/edu0000018 is OK
- 10.1016/j.learninstruc.2006.02.005 is OK
- 10.1007/978-3-642-84010-4_1 is OK
- 10.21105/jose.00021 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by labarba** [comment]
:loudspeaker:  New Reviewer Sign-up Form!
https://forms.gle/Rizd3TcHnQKhrbYY7
